pkg/webhook/validatingwebhook/validate.go

```diff
@@ -0,0 +1,147 @@
+package validatingwebhook
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+
+	userv1 "github.com/openshift/api/user/v1"
+	runtimeClient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/pkg/errors"
+	rbac "k8s.io/api/rbac/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "k8s.io/api/admission/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var (
+	runtimeScheme = runtime.NewScheme()
+	codecs        = serializer.NewCodecFactory(runtimeScheme)
+	deserializer  = codecs.UniversalDeserializer()
+
+	log = logf.Log.WithName("users_rolebindings_validating_webhook")
+)
+
+type Validator struct {
+	Client runtimeClient.Client
+}
+
+func (v Validator) HandleValidate(w http.ResponseWriter, r *http.Request) {
+	var respBody []byte
+	body, err := ioutil.ReadAll(r.Body)
+	defer func() {
+		if err := r.Body.Close(); err != nil {
+			log.Error(err, "unable to close the body")
+		}
+	}()
+	if err != nil {
+		log.Error(err, "unable to read the body of the request")
+		w.WriteHeader(http.StatusInternalServerError)
+		respBody = []byte(fmt.Sprintf("unable to read the body of the request: %s", err))
+	} else {
+		// validate the request
+		respBody = validate(body, v.Client)
+		w.WriteHeader(http.StatusOK)
+	}
+	if _, err := w.Write(respBody); err != nil {
+		log.Error(err, "unable to write response")
+	}
+}
+
+func validate(body []byte, client runtimeClient.Client) []byte {
+	admReview := v1.AdmissionReview{}
+	if _, _, err := deserializer.Decode(body, nil, &admReview); err != nil {
+		log.Error(err, "unable to deserialize the admission review object", "body", body)
+		return denyAdmissionRequest(admReview, errors.Wrapf(err, "unable to deserialize the admission review object - body: %v", body))
+	}
+	// let's unmarshal the object to be sure that it's a rolebinding
+	rb := rbac.RoleBinding{}
+	if err := json.Unmarshal(admReview.Request.Object.Raw, &rb); err != nil || rb.Kind != "RoleBinding" {
+		if err == nil {
+			err = fmt.Errorf("request Object is not a rolebinding")
+		}
+		log.Error(err, "unable unmarshal rolebinding json object", "AdmissionReview", admReview)
+		return denyAdmissionRequest(admReview, errors.Wrapf(err, "unable to unmarshal object or object is not a rolebinding - raw request object: %v", admReview.Request.Object.Raw))
+	}
+	requestingUsername := admReview.Request.UserInfo.Username
+	// allow admission request if the user is a system user
+	if strings.HasPrefix(requestingUsername, "system:") {
+		return allowAdmissionRequest(admReview)
+	}
+	subjects := rb.Subjects
+	allServiceAccountsSubject := rbac.Subject{
+		Kind:     "Group",
+		Name:     "system:serviceaccounts",
```

Comment:
Can you also check if we need to block `system:serviceaccounts:` subjects too. With `:` in the end.